core/src/main/scala/org/bykn/bosatsu/Evaluation.scala

```diff
@@ -467,8 +476,13 @@ case class Evaluation(pm: PackageMap.Inferred, externals: Externals) {
          val inner = recurse((p, Right(expr)))._1
 
          inner.asLambda(name)
-       case Let(arg, e, in, _) =>
-         val eres = recurse((p, Right(e)))._1
+       case Let(arg, e, in, rec, _) =>
+         val eres =
+           if (rec.isRecursive) {
+             Scoped.recursive(arg,
```

Comment:
if recursive isn't call by name, we cal pull the recurse call up above and not repeat ourselves.